crates/runtime/src/traphandlers.rs

```diff
@@ -501,7 +501,7 @@ impl<'a> CallThreadState<'a> {
     fn update_stack_limit(&self, max_wasm_stack: usize) -> Result<impl Drop + '_, Trap> {
         // Make an "educated guess" to figure out where the wasm sp value should
         // start trapping if it drops below.
```

Comment:
Since I just looked at this: "educated guess" makes me a little bit nervous, given the importance of this :) Can you talk a bit about the implications of this, both in terms of security and, if any, resource use limits?

If no implications for either exists, then I think it'd at least be good to update the comment and explain why this is all fine.

Reply:
Ah good point! I've updated the comment. Now that we're using something like `psm` it's more official and less educated. Otherwise this just refers to the imprecision about accounting for wasm stacks. If you ask for wasm to have a megabyte of stack, we may end up giving it a megabyte plus or minus a few hundred bytes.

Reply:
Great comment, thank you!